node/libs/protobuf/src/proto_fmt.rs

```diff
@@ -285,6 +285,26 @@ pub fn read_optional<T: ProtoFmt>(field: &Option<T::Proto>) -> anyhow::Result<Op
     field.as_ref().map(ProtoFmt::read).transpose()
 }
 
+/// Parses a repeated proto struct into a map.
+pub fn read_map<T, K, V, F, G>(items: &[T], k: F, v: G) -> anyhow::Result<BTreeMap<K, V>>
```

Comment:
please document what these arguments are.
nit: you can make the signature more readable by using `impl Fn(&T) -> ...` instead of F and G.